### PR TITLE
make -> just migration recipe

### PR DESCRIPTION
**Pull request by sigma**
**Description**

This change proposes a migration path from current Makefile to future
justfile.

It introduces common modules for just recipes, an include file for
newly-converted Makefiles, and an example of usage for op-batcher.

This enables a smooth transition, keeping "make" working (but
advertising the new workflow) as long as we need it to while
introducing just wherever convenient.

**Metadata**

- part of #12686

### CODE REVIEW

**Comment by sigma** [comment]
Note: this is mostly a POC / RFC at this point. If there's buy-in, we can start migrating targets more seriously.

File: just/git.just [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import 'default.just'
+
+# Set default values for git info
+GITCOMMIT := env('GITCOMMIT', `git rev-parse HEAD 2> /dev/null || true`)
+GITDATE := env('GITDATE', `git show -s --format='%ct' 2> /dev/null|| true`)
+
+# Find version tag, prioritizing non-rc release tags
+VERSION := env('VERSION', `tags=$(git tag --points-at $GITCOMMIT 2> /dev/null|| true | grep '^op-batcher/' | sed 's/op-batcher\///' | sort -V)
```
Comment (sigma) [inline just/git.just]: oops, I missed the tag filtering logic.  Leaving it aside for now just to collect feedback. But this obviously needs to be fixed

Comment (sigma) [reply just/git.just]: that's fixed now

**Comment by tynes** [comment]
This is sick!

**Review by mslipper — APPROVED** [review]
LGTM. Really glad we're deprecating Make. What's the plan to communicate this, roll it out, and completely deprecate Make?

**Review by protolambda — APPROVED** [review]
Nice work on the `just` utils :+1: 

If we create draft PRs for docs updates, then I think we can migrate a service and merge the docs PR around the same time. The `just/deprecated.mk` is already very helpful.

**Comment by sigma** [comment]
I've thought of some issue: command-line variables for the make invocation are not passed properly to just right now, which is problematic and might disrupt some workflows. Let me take a stab at sanitizing MAKEFLAGS and injecting the result into the just invocation

**Comment by sigma** [comment]
> I've thought of some issue: command-line variables for the make invocation are not passed properly to just right now, which is problematic and might disrupt some workflows. Let me take a stab at sanitizing MAKEFLAGS and injecting the result into the just invocation

Alright, this is now fixed (and was indeed causing divergence of behavior in CircleCI)
side-note: I hate the makefile language 😛

**Pull request merged**